src/level_gen.cpp

```diff
@@ -63,23 +65,28 @@ static inline void populateExpertTrajectory(Engine &ctx, const Entity &agent, co
     }
 
     for(CountT i = agentInit.numPositions - 2; i >=0; i--)
-    {   
+    {
         if(!trajectory.valids[i] || !trajectory.valids[i+1])
         {
             trajectory.inverseActions[i] = Action{.acceleration = 0, .steering = 0, .headAngle = 0};
+            trajectory.inverseDeltaActions[i] = DeltaAction{.dx = 0, .dy = 0, .dyaw = 0};
             continue;
         }
         Rotation rot = Quat::angleAxis(trajectory.headings[i], madrona::math::up);
+        Position pos = Vector3{.x = trajectory.positions[i].x - ctx.data().mean.x, .y = trajectory.positions[i].y - ctx.data().mean.y, .z = 1};
```

Comment:
You should not be subtracting the means again. This will give you wrong positions. We are already subtracting the means above line 61. 

```suggestion
        Position pos = Vector3{.x = trajectory.positions[i].x, .y = trajectory.positions[i].y, .z = 1};
```